module/applications/actor/base-sheet.mjs

```diff
@@ -1112,7 +1112,7 @@ export default class ActorSheet5e extends ActorSheetMixin(ActorSheet) {
     // Case 2: Drop a leveled spell in a section without a mode.
     else if ( (level === "0") || !preparationMode ) {
       if ( this.document.type === "npc" ) {
-        prep.mode = this.document.system.details.spellLevel ? "prepared" : "innate";
+        prep.mode = this.document.system.attributes.spell.level ? "prepared" : "innate";
```

Comment:
We have optional chaining above. Maybe not necessary here.

```suggestion
        prep.mode = this.document.system.attributes.spell?.level ? "prepared" : "innate";
```

Reply:
Since we've already checked that this is a NPC actor, we know that `attributes.spell` object will exist so we can skip the chaining.